### ISSUE TITLE
Oauth logged in user sign out inconsistency issue

### ISSUE THREAD

**Issue by raviSussol** [openmsupply/msupply-dashboard#46]
Signing out after logged in via Oauth doesn't function properly. It re-directs back to the home page from the sign out session after few seconds when signing out. And if you able to click logout button during a time then nothing works - takes you back to the home page anyway. However, if you sign it out for second time then it works. See gif:

[Clip](https://recordit.co/QEjOUsjpUI)

**Comment by mark-prins** [comment]
Yeah, it's a bit of a pain. It's due to the two cookies in play - grafana is keeping a cookie and so is mSupply; signing out of grafana will clear that auth cookie, and redirect you to the mSupply oAuth login. That cookie is still valid, so I have that login screen redirect you back to grafana, which accepts the incoming connection and signs you in.

depending on timing, the logout button works ( it clears the mSupply cookie ) but clearly isn't ideal / good enough for clients.

Will need a bit of a refactor in the authentication process
